app/controllers/users.js

```diff
@@ -29,17 +29,132 @@ exports.authCallback = (req, res) => {
 
 exports.findUsers = (req, res) => {
   User.find({}).select('name email').then((allUsers) => {
-    res.send(allUsers);
+    res.status(200)
+      .json(allUsers);
   });
 };
 
 exports.findUser = (req, res) => {
   const userid = req.params.userid;
   User.findById(userid, (err, oneUser) => {
     if (!err) {
-      res.send(oneUser);
+      res.status(200)
+        .json(oneUser);
     } else {
-      res.send('An error occurred');
+      res.status(401)
```

Comment:
`401` means unauthorised which I am not sure is ideal in this case. You'll probably want to use a `400` or `500` in most cases for errors depending on what the case may be and a `404` if a particular resource is not found.

Tip: mongoose `findById` does not throw an error if the resource returned is empty. i.e `oneUser` in this case may still be null if `err` is null.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html